commit e376267dda6972ffc2eeea215901f08145e3c993
Author: Christian Mollekopf <chrigi_1@fastmail.fm>
Date:   Thu May 8 10:42:54 2014 +0200

    IMAP-Resource: Don't crash if IDLE collection is not subscribed locally.
    
    BUG: 329805

--- a/resources/imap/imapresource.cpp
+++ b/resources/imap/imapresource.cpp
@@ -645,6 +645,11 @@
   if ( !m_pool->serverCapabilities().contains( QLatin1String("IDLE") ) )
     return;
 
+  //Without password we don't even have to try
+  if (Settings::self()->password().isEmpty()) {
+    return;
+  }
+
   const QStringList ridPath = Settings::self()->idleRidPath();
   if ( ridPath.size() < 2 )
     return;
@@ -672,21 +677,17 @@
 
 void ImapResource::onIdleCollectionFetchDone( KJob *job )
 {
-  if ( job->error() == 0 ) {
-    Akonadi::CollectionFetchJob *fetch = static_cast<Akonadi::CollectionFetchJob*>( job );
-    Akonadi::Collection c = fetch->collections().first();
-
-    const QString password = Settings::self()->password();
-    if ( password.isEmpty() )
-      return;
-
-    ResourceStateInterface::Ptr state = ::ResourceState::createIdleState( this, c );
-    m_idle = new ImapIdleManager( state, m_pool, this );
-
-  } else {
+  if (job->error()) {
     kWarning() << "CollectionFetch for idling failed."
                << "error=" << job->error()
                << ", errorString=" << job->errorString();
+    return;
+  }
+  Akonadi::CollectionFetchJob *fetch = static_cast<Akonadi::CollectionFetchJob*>(job);
+  //Can be empty if collection is not subscribed locally
+  if (!fetch->collections().isEmpty()) {
+    ResourceStateInterface::Ptr state = ::ResourceState::createIdleState( this, fetch->collections().first() );
+    m_idle = new ImapIdleManager( state, m_pool, this );
   }
 }
 
